services/release/release.go

```diff
@@ -72,13 +72,17 @@ func createTag(gitRepo *git.Repository, rel *models.Release, msg string) (bool,
 			created = true
 			rel.LowerTagName = strings.ToLower(rel.TagName)
 
+			commits := repository.NewPushCommits()
```

Comment:
It's surprising that `NewPushCommits` only creates an empty struct ....

And the code below duplicates. Could we introduce some more meaningful functions?

Reply:
How would you deduplicate it? There may be a method which sets the head commit in the "constructor" but the compare url needs too many parameters. Feel free to change it.

Reply:
How about:

```
func NewPushCommitsForNotification(commit) {
	commits := repository.NewPushCommits()
	commits.HeadCommit = repository.CommitToPushCommit(commit)
	commits.CompareURL = rel.Repo.ComposeCompareURL(git.EmptySHA, commit.ID.String())
	return commits
}
```

Reply:
But that name would be wrong for the "deleted ref" notification or `push.go@200`.